MediaBrowser.Model/Dlna/StreamInfo.cs

```diff
@@ -618,201 +620,296 @@ public string GetOption(string name)
+                    sb.Append(CopyTimestamps.ToString(CultureInfo.InvariantCulture).ToLowerInvariant());
                 }
 
-                list.Add(new NameValuePair("RequireAvc", item.RequireAvc.ToString(CultureInfo.InvariantCulture).ToLowerInvariant()));
+                sb.Append("&RequireAvc=");
+                sb.Append(RequireAvc.ToString(CultureInfo.InvariantCulture).ToLowerInvariant());
             }
 
-            list.Add(new NameValuePair("Tag", item.MediaSource.ETag ?? string.Empty));
-
-            string subtitleCodecs = item.SubtitleCodecs.Length == 0 ?
-               string.Empty :
-               string.Join(",", item.SubtitleCodecs);
-
-            list.Add(new NameValuePair("SubtitleCodec", item.SubtitleStreamIndex.HasValue && item.SubtitleDeliveryMethod == SubtitleDeliveryMethod.Embed ? subtitleCodecs : string.Empty));
-
-            if (isHls)
+            var etag = MediaSource?.ETag;
+            if (!string.IsNullOrEmpty(etag))
             {
-                list.Add(new NameValuePair("SegmentContainer", item.Container ?? string.Empty));
-
-                if (item.SegmentLength.HasValue)
-                {
-                    list.Add(new NameValuePair("SegmentLength", item.SegmentLength.Value.ToString(CultureInfo.InvariantCulture)));
-                }
-
-                if (item.MinSegments.HasValue)
-                {
-                    list.Add(new NameValuePair("MinSegments", item.MinSegments.Value.ToString(CultureInfo.InvariantCulture)));
-                }
+                sb.Append("&Tag=");
+                sb.Append(etag);
+            }
 
-                list.Add(new NameValuePair("BreakOnNonKeyFrames", item.BreakOnNonKeyFrames.ToString(CultureInfo.InvariantCulture)));
+            if (SubtitleStreamIndex.HasValue && SubtitleDeliveryMethod == SubtitleDeliveryMethod.Embed && SubtitleCodecs.Length != 0)
+            {
+                sb.Append("&SubtitleCodec=");
+                sb.AppendJoin(',', SubtitleCodecs);
             }
 
-            foreach (var pair in item.StreamOptions)
+            foreach (var pair in StreamOptions)
             {
                 if (string.IsNullOrEmpty(pair.Value))
                 {
                     continue;
                 }
 
                 // strip spaces to avoid having to encode h264 profile names
-                list.Add(new NameValuePair(pair.Key, pair.Value.Replace(" ", string.Empty, StringComparison.Ordinal)));
+                sb.Append('&');
+                sb.Append(pair.Key);
+                sb.Append('=');
+                sb.Append(pair.Value.Replace(" ", string.Empty, StringComparison.Ordinal));
             }
 
-            if (!item.IsDirectStream)
+            if (!IsDirectStream)
             {
-                list.Add(new NameValuePair("TranscodeReasons", item.TranscodeReasons.ToString()));
+                sb.Append("&TranscodeReasons=");
+                sb.AppendJoin(',', TranscodeReasons.ToString());
```

Comment:
`.ToString()` on the flags enum returns something like `"AudioIsExternal, DirectPlayError"` so this doesn't work as intended

Reply:
Should work now as intended